sdk/core/System.ServiceModel.Rest/src/Pipeline/HttpPipelineMessageTransport.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Net.Http;
+using System.ServiceModel.Rest.Internal;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ServiceModel.Rest.Core.Pipeline;
+
+// Introduces the dependency on System.Net.Http;
+
+public partial class HttpPipelineMessageTransport : PipelineTransport<PipelineMessage>, IDisposable
+{
+    private readonly HttpClient _httpClient;
+
+    // TODO: remove this when refactor is complete.
+    public HttpClient Client => _httpClient;
+
+    private bool _disposed;
+
+    public HttpPipelineMessageTransport() : this(CreateDefaultClient())
+    {
+    }
+
+    public HttpPipelineMessageTransport(HttpClient client)
+    {
+        ClientUtilities.AssertNotNull(client, nameof(client));
+
+        _httpClient = client;
+    }
+
+    private static HttpClient CreateDefaultClient()
+    {
+        // TODO:
```

Comment:
Let's validate that this is the feature set we want to separate out from Azure.Core, e.g. Azure.Core adds the items on the list below and they're not in this lib.